test/parallel/test-util.js

```diff
@@ -86,7 +86,10 @@ assert.deepEqual(util._extend({a:1, b:2}, {b:3}), {a:1, b:3});
 
 // inherits
 var ctor = function() {};
+var superCtor = function() {};
+class A {}
 assert.throws(function() { util.inherits(ctor, {}); }, TypeError);
 assert.throws(function() { util.inherits(ctor, null); }, TypeError);
-assert.throws(function() { util.inherits(null, ctor); }, TypeError);
-assert.doesNotThrow(function() { util.inherits(ctor, ctor); }, TypeError);
+assert.throws(function() { util.inherits(null, superCtor); }, TypeError);
+assert.doesNotThrow(function() { util.inherits(ctor, superCtor); }, TypeError);
```

Comment:
Reason of this change:
```
> function A(){}
undefined
> util.inherits(A, A)
TypeError: Cyclic __proto__ value
    at Function.setPrototypeOf (native)
    at Object.exports.inherits (util.js:764:10)
...
```

It doesn't make sense to make a function inherit from itself, but I guess it's still a semver-major change?